src/components/Filters.js

```diff
@@ -109,6 +112,96 @@ export default class Filters {
+
+    for (let i = 0; i < filtersChildren.length; i++) {
+      const element = filtersChildren[i];
+      const filterName = element.querySelector('.mt-filter-name').value;
+      const columnDetails = this.maptable.columnDetails[filterName];
+      const filterOutput = [columnDetails.filterMethod];
+      if (columnDetails.filterMethod === 'compare') {
+        const filterRangeSelect = element.querySelector('.mt-filter-range');
+        filterOutput[1] = filterRangeSelect.value;
+        if (filterRangeSelect.value !== 'any') {
+          if (filterRangeSelect.value === 'BETWEEN') {
+            const filterValueMin = element.querySelector('.mt-filter-value-min').value;
+            const filterValueMax = element.querySelector('.mt-filter-value-max').value;
+            if (filterValueMin !== '' && filterValueMax === '') {
+              filterOutput[2] = filterValueMin;
+              filterOutput[3] = filterValueMax;
+            }
+          } else {
+            const filterValue = element.querySelector('.mt-filter-value-min').value;
+            filterOutput[2] = filterValue;
+          }
+        }
+      } else if (columnDetails.filterMethod === 'field' ||
+        columnDetails.filterMethod === 'dropdown') {
+        filterOutput[1] = '';
+        const filterValue = element.querySelector('.mt-filter-value').value;
+        filterOutput[2] = filterValue;
+      }
+      if (filterOutput[1] !== 'any' && filterOutput[2] && filterOutput[2] !== '') {
+        output[filterName] = filterOutput;
+      }
+    }
+    return output;
+  }
+
+  setCriteria(criteria) {
+    Object.keys(criteria).forEach(filterName => {
+      this.add();
+      const criterion = criteria[filterName];
+      const row = document
+      .querySelector(`#mt-filters-elements [data-mt-filter-name="${filterName}"]`);
+      if (row) {
+        if (criterion[0] === 'compare') {
+          row.querySelector('.mt-filter-range').value = criterion[1];
+          if (criterion[1] !== 'any') {
+            if (criterion[1] === 'BETWEEN') {
+              row.querySelector('.mt-filter-value-min').value = criterion[2];
+              row.querySelector('.mt-filter-value-max').value = criterion[3];
+            } else {
+              row.querySelector('.mt-filter-value-min').value = criterion[2];
+            }
+          }
+        } else if (criterion[0] === 'field' || criterion[0] === 'dropdown') {
+          row.querySelector('.mt-filter-value').value = criterion[2];
+        }
+      }
+    });
+  }
+
+  restoreState() {
+    this.restoringState = true;
+    const params = document.location.href.split('!mt-filters=');
+    const defaultCriteria = (params[1]) ? params[1].split('!mt')[0] : null;
+    if (defaultCriteria) {
+      try {
+        this.setCriteria(JSON.parse(defaultCriteria));
+      } catch (e) {
+        console.log(`Maptable: Invalid URL State for mt-filters ${e.message}`);
```

Comment:
`throw` is a blocking statement. And we don't want to block the JS execution if the URL state is wrong.